soccer/gameplay/plays/defense/pileup.py

```diff
@@ -0,0 +1,125 @@
+import robocup
+import constants
+import play
+import skills
+import tactics
+import main
+import evaluation
+import math
+import random
+
+'''
+def fighting(r, x, y):    
+    d = sqrt(abs(x - r.pos.x) + abs(y - r.pos.y))
+    return d <= (3 * constants.Robot.Radius + constants.Ball.Radius)
+'''
+
+class Pileup(play.Play):
+
+    def __init__(self):
+        super().__init__(continuous=False)
+        
+        x = main.ball().pos.x #x-coordinate of ball's position
+        y = main.ball().pos.y #y-coordinate of ball's position
+        us = main.our_robots()
+        them = main.their_robots()
+        width = constants.Field.Width
+        base_width = -(width / 2)
+        length = constants.Field.Length
+        #print((length * 5)/6)
+        print(base_width, base_width + ((width * 2)/3))
+
+        ''' THIS IS TO BE IMPLEMENTED NEXT. GOAL IS TO CALCULATE AVERAGE POSITION OF OUR BOTS AND THEIR BOTS TO DETERMINE WHERE TO PLACE COLLECTING ROBOTS IN THE EVENT OF A GIVEN PILEUP
+
+        combatants = [] #robots that are actively fighting for the ball at the moment
+        #directions = []
+        for u in us:
+            if fighting(u, x, y):
+                combatants.add(u)
+            print(u)
+            #directions.append(u.angle)
+            #print("Angle: ", u.angle)
+        for t in them:
+            if fighting(t, x, y):
+                combatants.add(t)
+            print(t)
+            #directions.append(t.angle)
+            #print("Angle: ", t.angle)
+        #avg = sum(directions) / len(directions)
+        #print("Average angle: ", avg)
+        #print(str(x) + "," + str(y))
+        print(len(combatants), " robots fighting for the ball")
+        self.standbyBot1 = None
+        self.standbyBot2 = None
+        '''
+       
+        their_end = (length * 7)/8
+        our_end = length/8
+        left = base_width + (width/3)
+        right = base_width + (width*2)/3
+
+        if (y >= their_end): 
+            if (x <= left): # left corner of attacking half
+                print("pileup in attacking left corner")
+                pos1 = robocup.Point(x, y - 1)
```

Comment:
You still wanna change the y-1, to something like y - (c * length) , where c is a constant, because depending on the field a -1 to the coordinates can mean different things

Reply:
done, added constants dist and half_dist (for pythagorean calculations)